t/parser.t

```diff
@@ -829,3 +829,318 @@ subtest 'sources limit params', sub {
     $test->('Checker', sub { $_[0]->{checker} });
     $test->('Interactor', sub { $_[0]->{interactor} });
 };
+
+subtest 'formal', sub {
+    plan tests => 27;
```

Comment:
Seems to be a lot of repetition in test case construction.
Some repetition is ok for tests, but try to extract maybe a couple of common parts.